src/status_im/native_module/impl/module.cljs

```diff
@@ -143,50 +144,20 @@
                            (callback r')))]
            (.callJail status jail-id (types/clj->json path) (types/clj->json params') cb))))))
 
-;; TODO(rasom): temporal solution, should be fixed on status-go side
-(def check-raw-calls-interval 400)
-(def interval-between-calls 100)
-;; contains all calls to jail before with duplicates
-(def raw-jail-calls (atom '()))
-;; contains only calls that passed duplication check
-(def jail-calls (atom '()))
-
-(defn remove-duplicate-calls
-  "Removes duplicates by [jail path] keys, remains the last one."
-  [[all-keys calls] {:keys [jail-id path] :as call}]
-  (if (contains? all-keys [jail-id path])
-    [all-keys calls]
-    [(conj all-keys [jail-id path])
-     (conj calls call)]))
-
-(defn check-raw-calls-loop!
-  "Only the last call with [jail path] key is added to jail-calls list
-  from raw-jail-calls"
-  []
-  (go-loop [_ nil]
-    (let [[_ new-calls] (reduce remove-duplicate-calls [#{} '()] @raw-jail-calls)]
-      (reset! raw-jail-calls '())
-      (swap! jail-calls (fn [old-calls]
-                           (concat new-calls old-calls))))
-    (recur (<! (timeout check-raw-calls-interval)))))
-
-(defn execute-calls-loop!
-  "Calls to jail are executed ne by one with interval-between-calls,
-  which reduces chances of response shuffling"
-  []
-  (go-loop [_ nil]
-    (let [next-call (last @jail-calls)]
-      (swap! jail-calls butlast)
-      (when next-call
-        (execute-call next-call)))
-    (recur (<! (timeout interval-between-calls)))))
+;; TODO(janherich): temporal solution, should be fixed on status-go side
```

Comment:
maybe it's already fixed on go side and we can remove this?

Reply:
@janherich yes. it was fixed https://github.com/status-im/status-go/issues/372, so we can remove this temporal solution

Reply:
@flexsurfer 
1. PR is still wip, idk guys why you started reviewing 
2. This code still looks useful because it increases the number of calls to the jail 

@janherich 
`dedupe` will not work for calls, because they all are different at least because of `:callback` property. We probably could write our own transducer here.

Reply:
@rasom Yes, I will do it, `(dedupe)` is a super quick solution, but we can't do the same thing as before (deduplicate on basis of `[jail-id path]` alone) because it partially broke subscriptions (in addition to another recent command-loading changes which totally broke subscriptions).
The reason why is that when you had more then one subscription dependent on common property (such as `feeValue` and `feeExplanation` dependent on `[:sliderValue])`, only one subscription call went through, all other were discarded by the de-duplication code previously there (because the `[jail-id path]` tule is the same for all subscriptions from one bot).
And once I removed de-duplication from the former code, I realised it performed jail calls in opposite order (when there was more then one "waiting") + never properly cleaned the "waiting queue" so I decided to re-write it to something much simpler which is doing the same thing.

Reply:
tbh to make it work properly with subscriptions, i mean old code, we just need to add one condition there. Just in case if writing of a proper solution with transducer will take too much time.

Reply:
@rasom I thought the same as well but it's not true, see the last paragraph of the comment above - once the de-duplication is working in a way that more then one call is "waiting to be executed" (after de-duplication) the calls are performed in wrong (opposite) order + there always seems to be one extra old call which is executed with the newest one.

Reply:
>  there always seems to be one extra old call which is executed with the newest one.
can't reproduce this, though ordering and `:subscription` problem can be fixed with this code
https://github.com/status-im/status-react/commit/3107a9936401c3db3117d8bd97b55bb11a3a0c69

the rest stuff is related to proper handling of the callback